docs/reference/search/search-your-data/learning-to-rank.asciidoc

```diff
@@ -0,0 +1,136 @@
+[[learning-to-rank]]
+== Learning To Rank
+
+preview::["The Learning To Rank feature is in technical preview and may be changed or removed in a future release. Elastic will work to fix any issues, but this feature is not subject to the support SLA of official GA features."]
+
+NOTE: This feature was introduced in version 8.12.0 and is only available to certain subscription levels.
+For more information, see {subscriptions}.
+
+Learning To Rank (LTR) uses a trained machine learning (ML) model to build a
+ranking function for your search engine. Typically, the model is used as a
+second stage re-ranker, to improve the relevance of search results returned by a
+simpler, first stage retrieval algorithm. The LTR function takes a list of
+documents and a search context and outputs ranked documents:
+
+[[learning-to-rank-overview-diagram]]
+.Learning To Rank overview
+image::images/search/learning-to-rank-overview.png[Learning To Rank overview,align="center"]
+
+
+[discrete]
+[[learning-to-rank-search-context]]
+=== Search context
+
+In addition to the list of documents to sort, the LTR function also requires a
+search context. Typically, this search context includes at least the search
+terms provided by the user (text_query in the example above).
+The search context can also provide additional information used in the ranking mode.
+This could be information about the user doing the search (like demographic data, geolocation, or age); about the query (like query length); or document in the context of the query (like score for the title field).
```

Comment:
```suggestion
This could be information about the user doing the search (such as demographic data, geolocation, or age); about the query (such as query length); or document in the context of the query (such as score for the title field).
```